In [2]:
import numpy as np
import xarray as xr
from datetime import datetime
import datetime as dt
import glob

## 1. Obtener un xarray dataset. Debe ser similar a datos preprocesados de algún satélite: landsat o sentinel2. Ahora se utilizan datos que no fueron registrados en la db del datacube

## 1b Se utilizan los datos de la ruta:

`/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data`

In [3]:
path = '/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data/' 

nc_files = [f for f in glob.glob(path + "/LS5*.nc", recursive=True)]

In [4]:
nc_files

['/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data/LS5_espa_21_-19_19960312164217000000.nc',
 '/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data/LS5_espa_21_-19_19960124163902000000.nc',
 '/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data/LS5_espa_21_-19_19961123165617000000.nc',
 '/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data/LS5_espa_21_-19_19951121163445000000.nc',
 '/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data/LS5_espa_21_-19_19950206164648000000.nc',
 '/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data/LS5_espa_21_-19_19950310164531000000.nc',
 '/LUSTRE/MADMEX/tasks/2019_tasks/generacion_datos_entrenamiento_cultivos/examples_source_data/LS5_espa_21_-19_19961107165532000000.nc']

In [5]:
ds_combined_no_chunks = xr.open_mfdataset(path +'/LS5*.nc',
                                          data_vars='minimal', coords='minimal',
                                          combine='by_coords')

Combine nc files

In [6]:
ds_combined_no_chunks

<xarray.Dataset>
Dimensions:   (time: 7, x: 1667, y: 1667)
Coordinates:
  * x         (x) float64 2.027e+06 2.027e+06 2.027e+06 ... 2.077e+06 2.077e+06
  * y         (y) float64 1.527e+06 1.526e+06 1.526e+06 ... 1.477e+06 1.477e+06
  * time      (time) datetime64[ns] 1995-02-06T16:46:48 ... 1996-11-23T16:56:17
Data variables:
    crs       int32 -2147483647
    blue      (time, y, x) float32 dask.array<chunksize=(1, 1667, 1667), meta=np.ndarray>
    green     (time, y, x) float32 dask.array<chunksize=(1, 1667, 1667), meta=np.ndarray>
    red       (time, y, x) float32 dask.array<chunksize=(1, 1667, 1667), meta=np.ndarray>
    nir       (time, y, x) float32 dask.array<chunksize=(1, 1667, 1667), meta=np.ndarray>
    swir1     (time, y, x) float32 dask.array<chunksize=(1, 1667, 1667), meta=np.ndarray>
    swir2     (time, y, x) float32 dask.array<chunksize=(1, 1667, 1667), meta=np.ndarray>
    pixel_qa  (time, y, x) float32 dask.array<chunksize=(1, 1667, 1667), meta=np.ndarray>
    datase

In [7]:
ds_combined_no_chunks.time

<xarray.DataArray 'time' (time: 7)>
array(['1995-02-06T16:46:48.000000000', '1995-03-10T16:45:31.000000000',
       '1995-11-21T16:34:45.000000000', '1996-01-24T16:39:02.000000000',
       '1996-03-12T16:42:17.000000000', '1996-11-07T16:55:32.000000000',
       '1996-11-23T16:56:17.000000000'], dtype='datetime64[ns]')
Coordinates:
  * time     (time) datetime64[ns] 1995-02-06T16:46:48 ... 1996-11-23T16:56:17
Attributes:
    standard_name:  time
    long_name:      Time, unix time-stamp
    axis:           T

In [8]:
ds_combined = ds_combined_no_chunks.chunk({'time': -1})

In [9]:
ds_combined

<xarray.Dataset>
Dimensions:   (time: 7, x: 1667, y: 1667)
Coordinates:
  * x         (x) float64 2.027e+06 2.027e+06 2.027e+06 ... 2.077e+06 2.077e+06
  * y         (y) float64 1.527e+06 1.526e+06 1.526e+06 ... 1.477e+06 1.477e+06
  * time      (time) datetime64[ns] 1995-02-06T16:46:48 ... 1996-11-23T16:56:17
Data variables:
    crs       int32 -2147483647
    blue      (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    green     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    red       (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    nir       (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    swir1     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    swir2     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    pixel_qa  (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    datase

In [10]:
ds_combined.blue.attrs

OrderedDict([('grid_mapping', 'crs'),
             ('units', 'reflectance'),
             ('long_name', 'Surface Reflectance 0.45-0.52 microns (Blue)'),
             ('alias', 'band_1')])

In [11]:
ds_combined.blue

<xarray.DataArray 'blue' (time: 7, y: 1667, x: 1667)>
dask.array<rechunk-merge, shape=(7, 1667, 1667), dtype=float32, chunksize=(7, 1667, 1667), chunktype=numpy.ndarray>
Coordinates:
  * x        (x) float64 2.027e+06 2.027e+06 2.027e+06 ... 2.077e+06 2.077e+06
  * y        (y) float64 1.527e+06 1.526e+06 1.526e+06 ... 1.477e+06 1.477e+06
  * time     (time) datetime64[ns] 1995-02-06T16:46:48 ... 1996-11-23T16:56:17
Attributes:
    grid_mapping:  crs
    units:         reflectance
    long_name:     Surface Reflectance 0.45-0.52 microns (Blue)
    alias:         band_1

In [12]:
ds_combined.blue.isel(time=3).values

array([[133., 133.,  18., ..., 211., 211., 154.],
       [ 75.,  47.,  75., ..., 211., 209., 154.],
       [133., 105.,  75., ..., 182., 182., 209.],
       ...,
       [ nan,  nan,  nan, ...,  nan,  nan,  nan],
       [ nan,  nan,  nan, ...,  nan,  nan,  nan],
       [ nan,  nan,  nan, ...,  nan,  nan,  nan]], dtype=float32)

## 2b Calcular serie de tiempo

Se utiliza la función `resample` para generar una serie de tiempo. Ver por ejemplo:

https://github.com/CONABIO/antares3-sandbox/blob/master/notebooks/time_series_computation_xarray/time_series_computation_xarray_3.ipynb

In [13]:
ds_combined

<xarray.Dataset>
Dimensions:   (time: 7, x: 1667, y: 1667)
Coordinates:
  * x         (x) float64 2.027e+06 2.027e+06 2.027e+06 ... 2.077e+06 2.077e+06
  * y         (y) float64 1.527e+06 1.526e+06 1.526e+06 ... 1.477e+06 1.477e+06
  * time      (time) datetime64[ns] 1995-02-06T16:46:48 ... 1996-11-23T16:56:17
Data variables:
    crs       int32 -2147483647
    blue      (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    green     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    red       (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    nir       (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    swir1     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    swir2     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    pixel_qa  (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    datase

In [14]:
ds_combined['ndvi'] = ((ds_combined.nir-ds_combined.red)/(ds_combined.nir+ds_combined.red))

In [15]:
ds_combined

<xarray.Dataset>
Dimensions:   (time: 7, x: 1667, y: 1667)
Coordinates:
  * x         (x) float64 2.027e+06 2.027e+06 2.027e+06 ... 2.077e+06 2.077e+06
  * y         (y) float64 1.527e+06 1.526e+06 1.526e+06 ... 1.477e+06 1.477e+06
  * time      (time) datetime64[ns] 1995-02-06T16:46:48 ... 1996-11-23T16:56:17
Data variables:
    crs       int32 -2147483647
    blue      (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    green     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    red       (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    nir       (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    swir1     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    swir2     (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    pixel_qa  (time, y, x) float32 dask.array<chunksize=(7, 1667, 1667), meta=np.ndarray>
    datase

In [16]:
ds_combined.time.values

array(['1995-02-06T16:46:48.000000000', '1995-03-10T16:45:31.000000000',
       '1995-11-21T16:34:45.000000000', '1996-01-24T16:39:02.000000000',
       '1996-03-12T16:42:17.000000000', '1996-11-07T16:55:32.000000000',
       '1996-11-23T16:56:17.000000000'], dtype='datetime64[ns]')

In [17]:
begin="1995-02-06"

In [18]:
begin_time = datetime.strptime(begin, '%Y-%m-%d')

In [19]:
ts_1 = ds_combined.ndvi.resample(time='80D', 
                                 loffset=begin_time - 
                                 datetime.utcfromtimestamp(ds_combined['time'][0].values.astype(datetime)*1e-9)).nearest().interpolate_na('time')

In [20]:
ts_1

<xarray.DataArray 'ndvi' (time: 9, y: 1667, x: 1667)>
dask.array<transpose, shape=(9, 1667, 1667), dtype=float32, chunksize=(9, 1667, 1667), chunktype=numpy.ndarray>
Coordinates:
  * time     (time) datetime64[ns] 1995-02-06 1995-04-27 ... 1996-11-07
  * x        (x) float64 2.027e+06 2.027e+06 2.027e+06 ... 2.077e+06 2.077e+06
  * y        (y) float64 1.527e+06 1.526e+06 1.526e+06 ... 1.477e+06 1.477e+06

In [21]:
ts_1['time']

<xarray.DataArray 'time' (time: 9)>
array(['1995-02-06T00:00:00.000000000', '1995-04-27T00:00:00.000000000',
       '1995-07-16T00:00:00.000000000', '1995-10-04T00:00:00.000000000',
       '1995-12-23T00:00:00.000000000', '1996-03-12T00:00:00.000000000',
       '1996-05-31T00:00:00.000000000', '1996-08-19T00:00:00.000000000',
       '1996-11-07T00:00:00.000000000'], dtype='datetime64[ns]')
Coordinates:
  * time     (time) datetime64[ns] 1995-02-06 1995-04-27 ... 1996-11-07
Attributes:
    standard_name:  time
    long_name:      Time, unix time-stamp
    axis:           T

In [22]:
times=np.datetime_as_string(ts_1['time'],unit='D')

In [23]:
times

array(['1995-02-06', '1995-04-27', '1995-07-16', '1995-10-04',
       '1995-12-23', '1996-03-12', '1996-05-31', '1996-08-19',
       '1996-11-07'], dtype='<U28')

In [24]:
dframe_ts_1 = ts_1.to_dataframe()

/home/madmex_user/.local/lib/python3.6/site-packages/dask/core.py:119: RuntimeWarning: divide by zero encountered in true_divide
  return func(*args2)


In [25]:
dframe_ts_1

ndvi
time       y         x                  
1995-02-06 1526525.0 2027385.0  0.390029
                     2027415.0  0.390029
                     2027445.0  0.390029
                     2027475.0  0.390029
                     2027505.0  0.390029
                     2027535.0  0.390029
                     2027565.0  0.390029
                     2027595.0  0.392562
                     2027625.0  0.354659
                     2027655.0  0.333333
                     2027685.0  0.349475
                     2027715.0  0.341421
                     2027745.0  0.392562
                     2027775.0  0.458697
                     2027805.0  0.484995
                     2027835.0  0.463816
                     2027865.0  0.478122
                     2027895.0  0.469679
                     2027925.0  0.501037
                     2027955.0  0.458697
                     2027985.0  0.504559
                     2028015.0  0.497164
                     2028045.0  0.442448
                     2028075.0  0.468026
                     2028105.0  0.342540
                     2028135.0  0.308581
                     2028165.0  0.335975
                     2028195.0  0.361118
                     2028225.0  0.383158
                     2028255.0  0.422477
...                                  ...
1996-11-07 1476545.0 2076495.0  0.468693
                     2076525.0  0.584862
                     2076555.0  0.574118
                     2076585.0  0.574118
                     2076615.0  0.560704
                     2076645.0  0.528560
                     2076675.0  0.551951
                     2076705.0  0.594625
                     2076735.0  0.578331
                     2076765.0  0.562710
                     2076795.0  0.561025
                     2076825.0  0.599600
                     2076855.0  0.645480
                     2076885.0  0.617174
                     2076915.0  0.581318
                     2076945.0  0.549724
                     2076975.0  0.480687
                     2077005.0  0.442509
                     2077035.0  0.458728
                     2077065.0  0.489395
                     2077095.0  0.506454
                     2077125.0  0.390215
                     2077155.0  0.275275
                     2077185.0  0.295469
                     2077215.0  0.438739
                     2077245.0  0.526520
                     2077275.0  0.526724
                     2077305.0  0.526724
                     2077335.0  0.626016
                     2077365.0  0.672973

[25010001 rows x 1 columns]

In [26]:
dframe_ts_1=dframe_ts_1.reset_index('time')

In [27]:
dframe_ts_1

time      ndvi
y         x                             
1526525.0 2027385.0 1995-02-06  0.390029
          2027415.0 1995-02-06  0.390029
          2027445.0 1995-02-06  0.390029
          2027475.0 1995-02-06  0.390029
          2027505.0 1995-02-06  0.390029
          2027535.0 1995-02-06  0.390029
          2027565.0 1995-02-06  0.390029
          2027595.0 1995-02-06  0.392562
          2027625.0 1995-02-06  0.354659
          2027655.0 1995-02-06  0.333333
          2027685.0 1995-02-06  0.349475
          2027715.0 1995-02-06  0.341421
          2027745.0 1995-02-06  0.392562
          2027775.0 1995-02-06  0.458697
          2027805.0 1995-02-06  0.484995
          2027835.0 1995-02-06  0.463816
          2027865.0 1995-02-06  0.478122
          2027895.0 1995-02-06  0.469679
          2027925.0 1995-02-06  0.501037
          2027955.0 1995-02-06  0.458697
          2027985.0 1995-02-06  0.504559
          2028015.0 1995-02-06  0.497164
          2028045.0 1995-02-06  0.442448
          2028075.0 1995-02-06  0.468026
          2028105.0 1995-02-06  0.342540
          2028135.0 1995-02-06  0.308581
          2028165.0 1995-02-06  0.335975
          2028195.0 1995-02-06  0.361118
          2028225.0 1995-02-06  0.383158
          2028255.0 1995-02-06  0.422477
...                        ...       ...
1476545.0 2076495.0 1996-11-07  0.468693
          2076525.0 1996-11-07  0.584862
          2076555.0 1996-11-07  0.574118
          2076585.0 1996-11-07  0.574118
          2076615.0 1996-11-07  0.560704
          2076645.0 1996-11-07  0.528560
          2076675.0 1996-11-07  0.551951
          2076705.0 1996-11-07  0.594625
          2076735.0 1996-11-07  0.578331
          2076765.0 1996-11-07  0.562710
          2076795.0 1996-11-07  0.561025
          2076825.0 1996-11-07  0.599600
          2076855.0 1996-11-07  0.645480
          2076885.0 1996-11-07  0.617174
          2076915.0 1996-11-07  0.581318
          2076945.0 1996-11-07  0.549724
          2076975.0 1996-11-07  0.480687
          2077005.0 1996-11-07  0.442509
          2077035.0 1996-11-07  0.458728
          2077065.0 1996-11-07  0.489395
          2077095.0 1996-11-07  0.506454
          2077125.0 1996-11-07  0.390215
          2077155.0 1996-11-07  0.275275
          2077185.0 1996-11-07  0.295469
          2077215.0 1996-11-07  0.438739
          2077245.0 1996-11-07  0.526520
          2077275.0 1996-11-07  0.526724
          2077305.0 1996-11-07  0.526724
          2077335.0 1996-11-07  0.626016
          2077365.0 1996-11-07  0.672973

[25010001 rows x 2 columns]

In [28]:
dframe_ts_1.index = [dframe_ts_1.index.get_level_values('y'),dframe_ts_1.index.get_level_values('x'),
                     dframe_ts_1.index.map('{0[0]}/{0[1]}'.format)]

In [29]:
dframe_ts_1

time      ndvi
y         x                                                 
1526525.0 2027385.0 1526525.0/2027385.0 1995-02-06  0.390029
          2027415.0 1526525.0/2027415.0 1995-02-06  0.390029
          2027445.0 1526525.0/2027445.0 1995-02-06  0.390029
          2027475.0 1526525.0/2027475.0 1995-02-06  0.390029
          2027505.0 1526525.0/2027505.0 1995-02-06  0.390029
          2027535.0 1526525.0/2027535.0 1995-02-06  0.390029
          2027565.0 1526525.0/2027565.0 1995-02-06  0.390029
          2027595.0 1526525.0/2027595.0 1995-02-06  0.392562
          2027625.0 1526525.0/2027625.0 1995-02-06  0.354659
          2027655.0 1526525.0/2027655.0 1995-02-06  0.333333
          2027685.0 1526525.0/2027685.0 1995-02-06  0.349475
          2027715.0 1526525.0/2027715.0 1995-02-06  0.341421
          2027745.0 1526525.0/2027745.0 1995-02-06  0.392562
          2027775.0 1526525.0/2027775.0 1995-02-06  0.458697
          2027805.0 1526525.0/2027805.0 1995-02-06  0.484995
          2027835.0 1526525.0/2027835.0 1995-02-06  0.463816
          2027865.0 1526525.0/2027865.0 1995-02-06  0.478122
          2027895.0 1526525.0/2027895.0 1995-02-06  0.469679
          2027925.0 1526525.0/2027925.0 1995-02-06  0.501037
          2027955.0 1526525.0/2027955.0 1995-02-06  0.458697
          2027985.0 1526525.0/2027985.0 1995-02-06  0.504559
          2028015.0 1526525.0/2028015.0 1995-02-06  0.497164
          2028045.0 1526525.0/2028045.0 1995-02-06  0.442448
          2028075.0 1526525.0/2028075.0 1995-02-06  0.468026
          2028105.0 1526525.0/2028105.0 1995-02-06  0.342540
          2028135.0 1526525.0/2028135.0 1995-02-06  0.308581
          2028165.0 1526525.0/2028165.0 1995-02-06  0.335975
          2028195.0 1526525.0/2028195.0 1995-02-06  0.361118
          2028225.0 1526525.0/2028225.0 1995-02-06  0.383158
          2028255.0 1526525.0/2028255.0 1995-02-06  0.422477
...                                            ...       ...
1476545.0 2076495.0 1476545.0/2076495.0 1996-11-07  0.468693
          2076525.0 1476545.0/2076525.0 1996-11-07  0.584862
          2076555.0 1476545.0/2076555.0 1996-11-07  0.574118
          2076585.0 1476545.0/2076585.0 1996-11-07  0.574118
          2076615.0 1476545.0/2076615.0 1996-11-07  0.560704
          2076645.0 1476545.0/2076645.0 1996-11-07  0.528560
          2076675.0 1476545.0/2076675.0 1996-11-07  0.551951
          2076705.0 1476545.0/2076705.0 1996-11-07  0.594625
          2076735.0 1476545.0/2076735.0 1996-11-07  0.578331
          2076765.0 1476545.0/2076765.0 1996-11-07  0.562710
          2076795.0 1476545.0/2076795.0 1996-11-07  0.561025
          2076825.0 1476545.0/2076825.0 1996-11-07  0.599600
          2076855.0 1476545.0/2076855.0 1996-11-07  0.645480
          2076885.0 1476545.0/2076885.0 1996-11-07  0.617174
          2076915.0 1476545.0/2076915.0 1996-11-07  0.581318
          2076945.0 1476545.0/2076945.0 1996-11-07  0.549724
          2076975.0 1476545.0/2076975.0 1996-11-07  0.480687
          2077005.0 1476545.0/2077005.0 1996-11-07  0.442509
          2077035.0 1476545.0/2077035.0 1996-11-07  0.458728
          2077065.0 1476545.0/2077065.0 1996-11-07  0.489395
          2077095.0 1476545.0/2077095.0 1996-11-07  0.506454
          2077125.0 1476545.0/2077125.0 1996-11-07  0.390215
          2077155.0 1476545.0/2077155.0 1996-11-07  0.275275
          2077185.0 1476545.0/2077185.0 1996-11-07  0.295469
          2077215.0 1476545.0/2077215.0 1996-11-07  0.438739
          2077245.0 1476545.0/2077245.0 1996-11-07  0.526520
          2077275.0 1476545.0/2077275.0 1996-11-07  0.526724
          2077305.0 1476545.0/2077305.0 1996-11-07  0.526724
          2077335.0 1476545.0/2077335.0 1996-11-07  0.626016
          2077365.0 1476545.0/2077365.0 1996-11-07  0.672973

[25010001 rows x 2 columns]

In [30]:
dframe_ts_1.index.names = ['y', 'x','y/x']

In [31]:
dframe_ts_1=dframe_ts_1.reset_index('y/x')

In [32]:
dframe_ts_1

y/x       time      ndvi
y         x                                                  
1526525.0 2027385.0  1526525.0/2027385.0 1995-02-06  0.390029
          2027415.0  1526525.0/2027415.0 1995-02-06  0.390029
          2027445.0  1526525.0/2027445.0 1995-02-06  0.390029
          2027475.0  1526525.0/2027475.0 1995-02-06  0.390029
          2027505.0  1526525.0/2027505.0 1995-02-06  0.390029
          2027535.0  1526525.0/2027535.0 1995-02-06  0.390029
          2027565.0  1526525.0/2027565.0 1995-02-06  0.390029
          2027595.0  1526525.0/2027595.0 1995-02-06  0.392562
          2027625.0  1526525.0/2027625.0 1995-02-06  0.354659
          2027655.0  1526525.0/2027655.0 1995-02-06  0.333333
          2027685.0  1526525.0/2027685.0 1995-02-06  0.349475
          2027715.0  1526525.0/2027715.0 1995-02-06  0.341421
          2027745.0  1526525.0/2027745.0 1995-02-06  0.392562
          2027775.0  1526525.0/2027775.0 1995-02-06  0.458697
          2027805.0  1526525.0/2027805.0 1995-02-06  0.484995
          2027835.0  1526525.0/2027835.0 1995-02-06  0.463816
          2027865.0  1526525.0/2027865.0 1995-02-06  0.478122
          2027895.0  1526525.0/2027895.0 1995-02-06  0.469679
          2027925.0  1526525.0/2027925.0 1995-02-06  0.501037
          2027955.0  1526525.0/2027955.0 1995-02-06  0.458697
          2027985.0  1526525.0/2027985.0 1995-02-06  0.504559
          2028015.0  1526525.0/2028015.0 1995-02-06  0.497164
          2028045.0  1526525.0/2028045.0 1995-02-06  0.442448
          2028075.0  1526525.0/2028075.0 1995-02-06  0.468026
          2028105.0  1526525.0/2028105.0 1995-02-06  0.342540
          2028135.0  1526525.0/2028135.0 1995-02-06  0.308581
          2028165.0  1526525.0/2028165.0 1995-02-06  0.335975
          2028195.0  1526525.0/2028195.0 1995-02-06  0.361118
          2028225.0  1526525.0/2028225.0 1995-02-06  0.383158
          2028255.0  1526525.0/2028255.0 1995-02-06  0.422477
...                                  ...        ...       ...
1476545.0 2076495.0  1476545.0/2076495.0 1996-11-07  0.468693
          2076525.0  1476545.0/2076525.0 1996-11-07  0.584862
          2076555.0  1476545.0/2076555.0 1996-11-07  0.574118
          2076585.0  1476545.0/2076585.0 1996-11-07  0.574118
          2076615.0  1476545.0/2076615.0 1996-11-07  0.560704
          2076645.0  1476545.0/2076645.0 1996-11-07  0.528560
          2076675.0  1476545.0/2076675.0 1996-11-07  0.551951
          2076705.0  1476545.0/2076705.0 1996-11-07  0.594625
          2076735.0  1476545.0/2076735.0 1996-11-07  0.578331
          2076765.0  1476545.0/2076765.0 1996-11-07  0.562710
          2076795.0  1476545.0/2076795.0 1996-11-07  0.561025
          2076825.0  1476545.0/2076825.0 1996-11-07  0.599600
          2076855.0  1476545.0/2076855.0 1996-11-07  0.645480
          2076885.0  1476545.0/2076885.0 1996-11-07  0.617174
          2076915.0  1476545.0/2076915.0 1996-11-07  0.581318
          2076945.0  1476545.0/2076945.0 1996-11-07  0.549724
          2076975.0  1476545.0/2076975.0 1996-11-07  0.480687
          2077005.0  1476545.0/2077005.0 1996-11-07  0.442509
          2077035.0  1476545.0/2077035.0 1996-11-07  0.458728
          2077065.0  1476545.0/2077065.0 1996-11-07  0.489395
          2077095.0  1476545.0/2077095.0 1996-11-07  0.506454
          2077125.0  1476545.0/2077125.0 1996-11-07  0.390215
          2077155.0  1476545.0/2077155.0 1996-11-07  0.275275
          2077185.0  1476545.0/2077185.0 1996-11-07  0.295469
          2077215.0  1476545.0/2077215.0 1996-11-07  0.438739
          2077245.0  1476545.0/2077245.0 1996-11-07  0.526520
          2077275.0  1476545.0/2077275.0 1996-11-07  0.526724
          2077305.0  1476545.0/2077305.0 1996-11-07  0.526724
          2077335.0  1476545.0/2077335.0 1996-11-07  0.626016
          2077365.0  1476545.0/2077365.0 1996-11-07  0.672973

[25010001 rows x 3 columns]

# 3 Compute features

`pip3.6 install --user tsfresh`

In [33]:
from tsfresh.feature_extraction import extract_features, MinimalFCParameters

In [34]:
extracted_features = extract_features(dframe_ts_1, column_id="y/x", column_sort="time",
                                      default_fc_parameters=MinimalFCParameters())

Feature Extraction: 100%|██████████| 50/50 [07:33<00:00,  9.06s/it]


In [35]:
extracted_features

variable             ndvi__length  ndvi__maximum  ndvi__mean  ndvi__median  \
id                                                                           
1476545.0/2027385.0           9.0       0.782884    0.692244      0.703608   
1476545.0/2027415.0           9.0       0.803297    0.742965      0.740030   
1476545.0/2027445.0           9.0       0.813833    0.763692      0.771824   
1476545.0/2027475.0           9.0       0.782884    0.729058      0.714496   
1476545.0/2027505.0           9.0       0.851969    0.727328      0.704835   
1476545.0/2027535.0           9.0       0.819737    0.710815      0.705534   
1476545.0/2027565.0           9.0       0.763385    0.696970      0.685824   
1476545.0/2027595.0           9.0       0.770807    0.714327      0.697941   
1476545.0/2027625.0           9.0       0.780863    0.709454      0.721506   
1476545.0/2027655.0           9.0       0.813833    0.718071      0.700901   
1476545.0/2027685.0           9.0       0.784856    0.710920      0.713932   
1476545.0/2027715.0           9.0       0.784856    0.714392      0.702670   
1476545.0/2027745.0           9.0       0.803297    0.687226      0.664899   
1476545.0/2027775.0           9.0       0.707778    0.620602      0.583294   
1476545.0/2027805.0           9.0       0.705530    0.641852      0.657734   
1476545.0/2027835.0           9.0       0.799267    0.736855      0.717496   
1476545.0/2027865.0           9.0       0.836320    0.766258      0.759124   
1476545.0/2027895.0           9.0       0.775064    0.718019      0.713932   
1476545.0/2027925.0           9.0       0.756594    0.663673      0.645427   
1476545.0/2027955.0           9.0       0.775570    0.680942      0.648432   
1476545.0/2027985.0           9.0       0.722341    0.656632      0.625205   
1476545.0/2028015.0           9.0       0.697867    0.617440      0.598418   
1476545.0/2028045.0           9.0       0.817957    0.682251      0.643431   
1476545.0/2028075.0           9.0       0.783765    0.690234      0.654730   
1476545.0/2028105.0           9.0       0.757597    0.683699      0.664887   
1476545.0/2028135.0           9.0       0.686740    0.648169      0.633910   
1476545.0/2028165.0           9.0       0.722517    0.678797      0.664283   
1476545.0/2028195.0           9.0       0.754556    0.710374      0.697390   
1476545.0/2028225.0           9.0       0.768934    0.702123      0.696400   
1476545.0/2028255.0           9.0       0.808236    0.737559      0.714819   
...                           ...            ...         ...           ...   
1526525.0/2076495.0           9.0       0.620690    0.554547      0.561937   
1526525.0/2076525.0           9.0       0.624017    0.564197      0.561937   
1526525.0/2076555.0           9.0       0.548431    0.473702      0.469498   
1526525.0/2076585.0           9.0       0.453985    0.355006      0.350030   
1526525.0/2076615.0           9.0       0.443038    0.370318      0.412779   
1526525.0/2076645.0           9.0       0.559245    0.458246      0.468034   
1526525.0/2076675.0           9.0       0.603068    0.535547      0.547143   
1526525.0/2076705.0           9.0       0.683782    0.648789      0.659981   
1526525.0/2076735.0           9.0       0.626582    0.593750      0.586538   
1526525.0/2076765.0           9.0       0.689756    0.611948      0.598695   
1526525.0/2076795.0           9.0       0.700127    0.631900      0.647913   
1526525.0/2076825.0           9.0       0.762336    0.670500      0.672574   
1526525.0/2076855.0           9.0       0.624906    0.572644      0.585909   
1526525.0/2076885.0           9.0       0.580742    0.482254      0.504087   
1526525.0/2076915.0           9.0       0.487347    0.420559      0.463316   
1526525.0/2076945.0           9.0       0.467589    0.425616      0.453985   
1526525.0/2076975.0           9.0       0.528196    0.461345      0.481382   
1526525.0/2077005.0           9.0       0.473514    0.442106      0.463316   
1526525.0/2077035.0   

** Obsérvese que disminuyó el número de renglones... dframe_ts_1 tiene 25,010,001 renglones y extracted_features tiene 2,778,889**